src/ctapipe/io/interpolating.py

```diff
@@ -0,0 +1,242 @@
+
+    def _read_parameter_table(self, tel_id):
+        pass
+
+
+class PointingInterpolator(Interpolator):
+    def __call__(self, tel_id, time):
+        """
+        Interpolate alt/az for given time and tel_id.
+
+        Parameters
+        ----------
+        tel_id : int
+            telescope id
+        time : astropy.time.Time
+            time for which to interpolate the pointing
+
+        Returns
+        -------
+        altitude : astropy.units.Quantity[deg]
+            interpolated altitude angle
+        azimuth : astropy.units.Quantity[deg]
+            interpolated azimuth angle
+        """
+
+        if tel_id not in self._interpolators:
+            if self.h5file is not None:
+                self._read_parameter_table(tel_id)
+            else:
+                raise KeyError(f"No table available for tel_id {tel_id}")
+
+        mjd = time.tai.mjd
+        az = u.Quantity(self._interpolators[tel_id](mjd), u.rad, copy=False)
+        alt = u.Quantity(self._secondary_interpolators[tel_id](mjd), u.rad, copy=False)
+        return alt, az
+
+    def add_table(self, tel_id, input_table):
+        """
+        Add a table to this interpolator
+
+        Parameters
+        ----------
+        tel_id : int
+            Telescope id
+        input_table : astropy.table.Table
+            Table of pointing values, expected columns
+            are ``time`` as ``Time`` column, ``azimuth`` and ``altitude``
+            as quantity columns for pointing and pointing correction data.
+        """
+
+        missing = {"time", "azimuth", "altitude"} - set(input_table.colnames)
+        if len(missing) > 0:
+            raise ValueError(f"Table is missing required column(s): {missing}")
+        for col in ("azimuth", "altitude"):
+            unit = input_table[col].unit
+            if unit is None or not u.rad.is_equivalent(unit):
+                raise ValueError(f"{col} must have units compatible with 'rad'")
+
+        if not isinstance(input_table["time"], Time):
+            raise TypeError("'time' column of pointing table must be astropy.time.Time")
+
+        az = input_table["azimuth"].quantity.to_value(u.rad)
+        # prepare azimuth for interpolation by "unwrapping": i.e. turning
+        # [359, 1] into [359, 361]. This assumes that if we get values like
+        # [359, 1] the telescope moved 2 degrees through 0, not 358 degrees
+        # the other way around. This should be true for all telescopes given
+        # the sampling speed of pointing values and their maximum movement speed.
+        # No telescope can turn more than 180° in 2 seconds.
+        az = np.unwrap(az)
+        alt = input_table["altitude"].quantity.to_value(u.rad)
+        mjd = input_table["time"].tai.mjd
+        self._interpolators[tel_id] = interp1d(mjd, az, **self.interp_options)
+        self._secondary_interpolators[tel_id] = interp1d(
+            mjd, alt, **self.interp_options
+        )
+
+    def _read_parameter_table(self, tel_id):
```

Comment:
This could probably be put into the base class, if we add a class attribute `table_path` this:

```
class PointingInterpolator(Interpolator):
    table_path = "/dl0/monitoring/telescope/pointing"

class PedestalInterpolator(Interpolator):
    table_path = "/dl0/monitoring/telescope/pedestal"
```